devlib/module/thermal.py

```diff
@@ -0,0 +1,108 @@
+#    Copyright 2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+
+from devlib.module import Module
+
+class TripPoint(object):
+    def __init__(self, zone, _id):
+        self._id = _id
+        self.zone = zone
+        self.temp_node = 'trip_point_' + _id + '_temp'
+        self.type_node = 'trip_point_' + _id + '_type'
+
+    def get_temperature(self):
+        """Returns the currently configured temperature of the trip point"""
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        return self.zone.target.read_value(temp_file)
+
+    def set_temperature(self, temperature):
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        self.zone.target_write_value(temp_file, temperature)
+
+    def get_type(self):
+        """Returns the type of trip point"""
+        type_file = self.zone.target.path.join(self.zone.path, self.type_node)
+        return self.zone.target.read_value(type_file)
+
+class ThermalZone(object):
+    def __init__(self, target, root, _id):
+        self.target = target
+        self.name = 'thermal_zone' + _id
+        self.path = target.path.join(root, self.name)
+        self.trip_points = {}
+
+        for entry in self.target.list_directory(self.path):
+            re_match = re.match('^trip_point_([0-9]+)_temp', entry)
+            if re_match is not None:
+                self.trip_point(re_match.group(1))
+
+    def trip_point(self, _id):
+        self.trip_points[int(_id)] = TripPoint(self, _id)
+
+    def is_enabled(self):
+        """Returns a boolean representing the 'mode' of the thermal zone"""
+        value = self.target.read_value(self.target.path.join(self.path, 'mode'))
+        if value == 'enabled':
+            return True
+        else:
+            return False
+
+    def set_mode(self, enable):
+        value = 'disable'
+        if enable:
+            value = 'enabled'
+
+        self.target.write_value(self.target.path.join(self.path, 'mode', value))
```

Comment:
Closing paren too far to the right. Should be

    self.target.write_value(self.target.path.join(self.path, 'mode'), value)